src/compiler/config/validate-dev-server.ts

```diff
@@ -28,24 +32,23 @@ export const validateDevServer = (config: d.Config, diagnostics: d.Diagnostic[])
 
   devServer.address = devServer.address.split('/')[0];
 
-  let addressPort: number;
   const addressSplit = devServer.address.split(':');
   if (addressSplit.length > 1) {
     if (!isNaN(addressSplit[1] as any)) {
       devServer.address = addressSplit[0];
-      addressPort = parseInt(addressSplit[1], 10);
+      devServer.port = parseInt(addressSplit[1], 10);
```

Comment:
Is it possible that we're accidentally prematurely overwriting this value if [`flags.port` is a number](https://github.com/ionic-team/stencil/pull/3335/files#diff-1b29641703a279611b48a9e91ce3ead0b080bc5a91df7b6347483b0231601257R43). It looks like if the previous implementation, `addressSplit[1]` took precedence over `flags.port`. Here, it looks like we've flipped that

Reply:
I think you're right - I moved the `if (isNumber(flags?.port)) { ... }` block up above this `addressSplit` business so that we'll start with `flags.port` and then overwrite it with `addressSplit[1]` if it makes sense to do so

Reply:
ok I went and re-did both the logic here and then also added a bunch of tests that passed with the original behavior (I filled in a few gaps with testing too, and de-duped some identical tests using `it.each` while I was in there).

Hopefully the new version 1) keeps the same behavior and 2) is easier to read and follow!